src/app/[slug]/page.jsx

```diff
@@ -0,0 +1,59 @@
+import { SinglePage } from '@/queries/pages'
+import { RichText } from '@graphcms/rich-text-react-renderer'
+import { notFound } from 'next/navigation'
+
+async function getPage(slug) {
+  const { page } = await fetch(process.env.HYGRAPH_ENDPOINT, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      query: SinglePage,
+      variables: { slug: slug }
+    })
+  })
+    .then((res) => res.json())
+    .then((res) => res.data)
+  return page
+}
+
+export async function generateMetadata({ params }) {
+  const page = await getPage(params.slug)
+  if (!page) return notFound()
+
+  return {
+    title: page?.seoOverride.title || page.title,
+    description: page.seo?.description || page.description,
+    openGraph: {
+      images: [
+        {
+          url: page?.seoOverride?.image?.url || page.coverImage?.url,
+          width: page?.seoOverride?.image?.width || page.coverImage?.width,
+          height: page?.seoOverride?.image?.height || page.coverImage?.height
+        }
+      ]
+    }
+  }
+}
+
+export default async function Page({ params }) {
+  const page = await getPage(params.slug)
+  return (
```

Comment:
I think it'd be good to check if the page exists here and return `notFound()` if not.

```ts
import { notFound } from 'next/navigation';

if(!page) {
  return notFound();
}
```

Probably on all pages.